sshkey/sshkey_test.go

```diff
@@ -0,0 +1,49 @@
+package sshkey
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bitrise-steplib/steps-activate-ssh-key/command"
+	"github.com/bitrise-steplib/steps-activate-ssh-key/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_WhenSSHKeyIsAdded_ThenItCallsSSHAddScript(t *testing.T) {
+	// Given
+	logger := log.NewDefaultLogger()
+
+	sshKeyPth := "ssh-key-path"
+	tmpDir := "temp-dir"
+	sshAddScriptPth := filepath.Join(tmpDir, addSSHKeyScriptFileName)
+
+	tempDirProvider := new(MockTempDirProvider)
+	tempDirProvider.On("CreateTempDir", mock.Anything).Return(tmpDir, nil).Once()
+
+	fileWriter := new(MockFileWriter)
+	fileWriter.On("Write", sshAddScriptPth, createAddSSHKeyScript(sshKeyPth), mock.Anything).Return(nil).Once()
+
+	cmd := new(MockCommand)
```

Comment:
If we don't extract these mocks I'd be happy to have a comment for them

// Setup default successful command

Reply:
Discussed we will get back to this question in the next PRs.